watertap/unit_models/steam_heater.py

```diff
@@ -0,0 +1,116 @@
+#################################################################################
+# WaterTAP Copyright (c) 2020-2024, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+
+from pyomo.environ import Var
+
+from idaes.core import (
+    declare_process_block_class,
+)
+from idaes.models.unit_models.heat_exchanger import HeatExchangerData
+from idaes.core.solvers import get_solver
+from idaes.core.util.model_statistics import degrees_of_freedom
+import idaes.logger as idaeslog
+
+
+_log = idaeslog.getLogger(__name__)
+
+
+__author__ = "Elmira Shamlou"
+
+
+@declare_process_block_class(
+    "FWHCondensing0D",
+    doc="""Feedwater Heater Condensing Section
+The feedwater heater condensing section model is a normal 0D heat exchanger
+model with an added constraint to calculate the steam flow such that the outlet
+of shell is a saturated liquid.""",
+)
+class FWHCondensing0DData(HeatExchangerData):
+    config = HeatExchangerData.CONFIG()
+
+    def build(self):
+        super().build()
+
+        @self.Constraint(
+            self.flowsheet().time,
+            self.hot_side.config.property_package.component_list,
+            doc="Mass balance",
+        )
+        def outlet_liquid_mass_balance(b, t, j):
+            b.hot_side.properties_out[t].flow_mass_phase_comp["Vap", j].fix(0)
+            return (
+                b.hot_side.properties_in[t].flow_mass_phase_comp["Vap", j]
+                + b.hot_side.properties_in[t].flow_mass_phase_comp["Liq", j]
+                == b.hot_side.properties_out[t].flow_mass_phase_comp["Liq", j]
+            )
+
+        units_meta = (
+            self.config.hot_side.property_package.get_metadata().get_derived_units
+        )
+
+        self.pressure_diff = Var(
+            self.flowsheet().time, initialize=0, units=units_meta("pressure")
+        )
+
+        self.pressure_diff.fix()
```

Comment:
I would also opt not to fix this to 0 like this. I think we can take a couple alternative approaches:
1) Don't unfix here, and require the user to fix this to 0 or some other value. On that note, do we want to enforce a domain for the variable to be NonNegativeReals? This would be in case the user tries entering a negative value (unsure whether this would be caught by something else).
2)  Add a config option such as `has_terminal_pressure_difference` with default set to False, which would in turn fix `pressure_diff` to 0 as you do here. Then, if user sets to True, they would then be responsible for fixing a value to `pressure_diff` to set up the square problem.

Reply:
@adam-a-a I'm concerned that this configuration option might unnecessarily complicate the model. The primary purpose of this steam heater is to adjust the IDAES heat exchanger's outlet so that full steam condensation occurs, which requires the outlet pressure to equal the saturation pressure.

Reply:
I get that, but I do not think we need to hard-code the outlet pressure to exactly equal saturation pressure. So, if adding a config option would be overcomplicating this, then option 1 sounds like the way to go. In reality, I wouldn't expect to target the outlet pressure to be exactly the same as the saturation pressure; there would be some overshoot, and I figure that this variable should be more transparent to the user so that he/she could specify the overpressure that they want to simulate.

Reply:
@adam-a-a I've implemented the second suggestion by incorporating a configuration option in the model. By default, this configuration sets the outlet pressure to match the saturation pressure, unless otherwise specified by the user.